### PR TITLE
outbound transportation form

### PR DESCRIPTION
**Pull request by FatimaSalami**
in this task outbound transp form was migrated
this needs to create time picker shared component
and add functions in date-helper because time shifts here 
so we need to fix time shift and format 
any question about functionality or certain code can be asked

### CODE REVIEW

File: src/lib/date-helper.js [hunk 1]
```diff
@@ -4,12 +4,47 @@ import dayjs from 'dayjs'
 // import moment from 'moment';
 import { compareAsc, format } from 'date-fns'
```
Comment (omarZrayka23) [inline src/lib/date-helper.js:5]: remove compareAsc

File: src/lib/date-helper.js [hunk 2]
```diff
@@ -53,7 +89,22 @@ function formatDateDefault(date) {
     'dateFormat'
   ]
   const timestamp = date instanceof Date ? date.getTime() : parseInt(date?.match(/\d+/)[0], 10)
-  const formattedDate = format(new Date(timestamp), formats)
+  const formattedDate = format(timeStamptoDate(timestamp), formats)
+
+  return formattedDate
+}
+
+function formatDateTimeDefault(date) {
```
Comment (omarZrayka23) [inline src/lib/date-helper.js:97]: can u make  formatDateTimeDefault  and formatDateDefault one function

File: src/lib/date-helper.js [hunk 3]
```diff
@@ -84,13 +135,12 @@ function getTimeInTimeZone(dateString, timeZone = 0) {
   return `${newHours}:${newMinutes}:${newSeconds}`
 }
 
-const formatDate = (dateString) => {
-  const date = new Date(dateString);
-  const timestamp = date.getTime();
-
-  return `/Date(${timestamp})/`;
-};
+const formatDate = dateString => {
```
Comment (omarZrayka23) [inline src/lib/date-helper.js:138]: is used formatDate ?? if not please clear

File: src/lib/date-helper.js [hunk 4]
```diff
@@ -53,7 +89,22 @@ function formatDateDefault(date) {
...
+function formatDateTimeDefault(date) {
+  if (!date) return
+
+  const formats = JSON.parse(window.localStorage.getItem('default') && window.localStorage.getItem('default'))[
+    'dateFormat'
+  ]
+
+  const timestamp = date instanceof Date ? date.getTime() : parseInt(date?.match(/\d+/)[0], 10)
+
+  const fullFormat = `${formats} hh:mm a`
```
Comment (omarZrayka23) [inline src/lib/date-helper.js:106]: how can controll time format ex : 2:30 pm , 14: 30 or 14:30:20

File: src/components/Inputs/CustomTimePicker.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
+import { useState } from 'react'
+
+import { InputAdornment, IconButton, Box } from '@mui/material'
```
Comment (omarZrayka23) [inline src/components/Inputs/CustomTimePicker.js:3]: clear Box

File: src/components/Inputs/CustomTimePicker.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+import { DISABLED, FORCE_ENABLED, HIDDEN, MANDATORY } from 'src/services/api/maxAccess'
+
+const CustomTimePicker = ({
+  name,
+  label,
+  value,
+  onChange,
+  error,
+  helperText,
```
Comment (omarZrayka23) [inline src/components/Inputs/CustomTimePicker.js:21]: clear helperText  if not used

File: src/components/Inputs/CustomTimePicker.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        disabled={disabled}
+        readOnly={_readOnly}
+        clearable
+        slotProps={{
+          textField: {
+            required: isRequired,
+            size: size,
+            fullWidth: fullWidth,
+            error: error,
+            helperText: helperText,
```
Comment (omarZrayka23) [inline src/components/Inputs/CustomTimePicker.js:81]: also clear helperText if not used

File: src/lib/date-helper.js (hunk 4 above)
Comment (FatimaSalami) [reply src/lib/date-helper.js:106]: we only used the format we need which is hh:mm a in outbound
i can't understand your question well

File: src/lib/date-helper.js (hunk 2 above)
Comment (FatimaSalami) [reply src/lib/date-helper.js:97]: why? one is when column needs to show date 
the other is when we need to show time with the date

File: src/components/Inputs/CustomTimePicker.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    (readOnly && accessLevel !== MANDATORY && accessLevel !== FORCE_ENABLED)
+
+  const _hidden = accessLevel ? accessLevel === HIDDEN : hidden
+
+  const isRequired = required || accessLevel === MANDATORY
+
+  return _hidden ? (
+    <></>
+  ) : (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
```
Comment (omarZrayka23) [inline src/components/Inputs/CustomTimePicker.js:54]: can u use AdapterDateFns??

File: src/lib/date-helper.js (hunk 2 above)
Comment (omarZrayka23) [reply src/lib/date-helper.js:97]: controlls by parameter

Comment (omarZrayka23) [reply src/lib/date-helper.js:97]: or put common parts of 2 functions in a function

File: src/lib/date-helper.js (hunk 1 above)
Comment (FatimaSalami) [reply src/lib/date-helper.js:5]: removed

File: src/lib/date-helper.js (hunk 2 above)
Comment (FatimaSalami) [reply src/lib/date-helper.js:97]: done in date helper

File: src/lib/date-helper.js (hunk 3 above)
Comment (FatimaSalami) [reply src/lib/date-helper.js:138]: i didn't add it; 
this needs to be checked in separate card

File: src/lib/date-helper.js (hunk 4 above)
Comment (FatimaSalami) [reply src/lib/date-helper.js:106]: formatDateTimeDefault do we pass param format to it?

**Comment by imtithalhalalo** [comment]
Screenshot 2024-09-30 152201
onClose is showing this error

File: src/lib/date-helper.js [hunk 9]
```diff
@@ -19,11 +51,12 @@ const formatDateFromApiInline = date => {
 }
 
 const formatDateToApi = date => {
-  const timestamp = date && date.valueOf()
+  const timestamp = date && dateToTimeStamp(new Date(date)).valueOf()
```
Comment (imtithalhalalo) [inline src/lib/date-helper.js:54]: why do you need dateToTimeStamp then new Date here?

**Comment by imtithalhalalo** [comment]
Screenshot 2024-09-30 152533
the name of the form is incorrect

File: src/pages/outbound-transportation/forms/OutboundTranspForm.js [hunk 10]
```diff
@@ -0,0 +1,571 @@
...
+        for (const order of filteredOrders) {
+          if (soIdSet.has(order.soId)) {
+            hasDuplicates = true
+            break
+          }
+          soIdSet.add(order.soId)
+        }
+
+        if (hasDuplicates) {
+          console.log('Duplicate soId found. Operation aborted.')
```
Comment (imtithalhalalo) [inline src/pages/outbound-transportation/forms/OutboundTranspForm.js:112]: is this console needed?

File: src/pages/outbound-transportation/forms/OutboundTranspForm.js [hunk 11]
```diff
@@ -0,0 +1,571 @@
...
+                    filter={!editMode ? item => item.activeStatus === 1 : undefined}
+                    name='dtId'
+                    label={labels.docType}
+                    readOnly={editMode}
+                    valueField='recordId'
+                    displayField='name'
+                    values={formik.values}
+                    onChange={async (event, newValue) => {
+                      formik.setFieldValue('dtId', newValue?.recordId || '')
+                      changeDT(newValue) //change REF??
```
Comment (imtithalhalalo) [inline src/pages/outbound-transportation/forms/OutboundTranspForm.js:361]: remove comment

File: src/pages/outbound-transportation/forms/OutboundTranspForm.js [hunk 12]
```diff
@@ -0,0 +1,571 @@
...
+        <Fixed>
+          <Grid container rowGap={1} xs={10} spacing={2}>
+            <Grid item xs={5}>
+              <CustomTextField
+                name='totalVolume'
+                label={labels.totVol}
+                maxAccess={maxAccess}
+                value={totalVol}
+                maxLength='30'
+                readOnly={true}
```
Comment (imtithalhalalo) [inline src/pages/outbound-transportation/forms/OutboundTranspForm.js:549]: pass just readOnly

File: src/lib/date-helper.js (hunk 4 above)
Comment (FatimaSalami) [reply src/lib/date-helper.js]: postponed until needed

**Comment by imtithalhalalo** [comment]
Screenshot 2024-09-30 153606
if they are required please add asterisk

File: src/lib/date-helper.js (hunk 9 above)
Comment (FatimaSalami) [reply src/lib/date-helper.js:54]: this function is added for time shifting keeping into consideration formats

File: src/pages/outbound-transportation/forms/OutboundTranspForm.js (hunk 10 above)
Comment (FatimaSalami) [reply src/pages/outbound-transportation/forms/OutboundTranspForm.js:112]: i kept it commented out

**Review by imtithalhalalo — APPROVED** [review]

**Pull request merged**